p2p/src/net/mock/backend.rs

```diff
@@ -55,24 +55,23 @@ use crate::{
             },
         },
         types::PubSubTopic,
-        Announcement,
+        Announcement, AsBannableAddress,
     },
 };
 
 #[derive(Debug)]
-struct PeerContext {
+struct PeerContext<T: TransportSocket> {
     _peer_id: MockPeerId,
+    address: T::Address,
     subscriptions: BTreeSet<PubSubTopic>,
     tx: mpsc::Sender<MockEvent>,
 }
 
 #[derive(Debug)]
-enum ConnectionState<T: TransportSocket> {
-    /// Connection established for outbound connection
-    OutboundAccepted { address: T::Address },
-
-    /// Connection established for inbound connection
-    InboundAccepted { address: T::Address },
+struct PendingPeerContext<T: TransportSocket> {
+    address: T::Address,
+    role: peer::Role,
+    tx: mpsc::Sender<MockEvent>,
 }
```

Comment:
A short doc comment describing what these structs are for would be nice (for `(Pending)PeerContext`)

Reply:
Sure, added.